### ISSUE TITLE
False positive on twitter links

### ISSUE THREAD

**Issue by robbak** [KilledMufasa/AmputatorBot#10]
On reddit, this bot is incorrectly being triggered by many twitter links, because of a tracking entry in the URL. For instance, this URL triggered in [this comment ](https://www.reddit.com/r/spacex/comments/iip8h3/rspacex_starlink11_launch_discussion_updates/g3cf0w9/)on r/spacex:
https://mobile.twitter.com/SpaceX/status/1300037857793290243?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Etweet

**Comment by unbeatable-101** [comment]
Maybe because the link says amp in it?

**Comment by KilledMufasa** [comment]
Apologies for my late response!

> Maybe because the link says amp in it?

Jup! Specifically, `amp%`, which is normally a pretty reliable way to check for AMP links. According to my data, AmputatorBot was able to find 1.922 links that way. 451 of those are mobile twitter links, which are, most of the time, false positives. However, removing the check for `amp%` would cause AmputatorBot to miss 1.471 AMP links in the process - not at option. 

The alternative, coding in a specific exception for mobile.twitter.com links for example, would be really error prune and high maintenance (because this is just one of many examples, and you never know when a domain changes their behaviour). There are already some checks in place to see if pages are AMP, but since AMP is implemented in widely different ways these checks still allow for stuff like this to happen.

To summarize, there really isn't much to do against these kind of false positives right now. I would rather have AmputatorBot take some wrong guesses every now and then than it missing actual AMP links. Thanks though for the feedback!
